### DISCUSSION TITLE
PIP-35: Comparison of communication protocols

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#201] [PIPs]
[PIP-35: Comparison of communication protocols](https://pips.pactus.org/PIPs/pip-35)

**Comment by b00f** [comment]
I suggest changing the title to reflect a comparison of messaging libraries.

We need to focus on identifying the following:  
- Which library is simpler.  
- Which library has a more mature developers
- Which library  a simple and comprehensive protocol.  
- Which library has fewer issues.  
- Which library has a native Go implementation and offers more support for native programming.  

It seems our options are: ZeroMQ, Nanomsg, and MQTT.

**Comment by b00f** [comment]
ZeroMQ and NanoMsg are broker less and therefore are suitable for peer-to-peer networks.
